File: frontend/src/types/evaluation.ts
export interface EvaluationItemResult {
  item_id: string
  semantic_score: number
  formality_score: number
  overall_score: number
  user_input: string
  reference_answer: string
  feedback_text: string
  suggested_phrases: string[]
}

export interface EvaluationResponse {
  session_id: string
  step: number
  overall_score: number
  items: EvaluationItemResult[]
  global_feedback: string
}

export interface Step3EvaluationResponse {
  session_id: string
  step: 3
  overall_score: number
  semantic_score: number
  formality_score: number
  coverage_percentage: number
  covered_points: string[]
  missing_points: string[]
  formality_issues: Array<{
    original: string
    suggested_text: string
    reason: string
  }>
  global_feedback: string
  annotated_text: string
}

File: backend/models/evaluation.py
from typing import Optional
from pydantic import BaseModel


class AnswerItem(BaseModel):
    item_id: str
    user_input: str


class EvaluationItemResult(BaseModel):
    item_id: str
    semantic_score: int
    formality_score: int
    overall_score: int
    user_input: str
    reference_answer: str
    feedback_text: str
    suggested_phrases: list[str]


class EvaluationResponse(BaseModel):
    session_id: str
    step: int
    overall_score: int
    items: list[EvaluationItemResult]
    global_feedback: str


class Step1EvaluateRequest(BaseModel):
    answers: list[AnswerItem]


class Step2EvaluateRequest(BaseModel):
    answers: list[AnswerItem]


class Step3EvaluateRequest(BaseModel):
    full_text: str


class Step3EvaluationResponse(BaseModel):
    session_id: str
    step: int = 3
    overall_score: int
    semantic_score: int
    formality_score: int
    coverage_percentage: int
    covered_points: list[str]
    missing_points: list[str]
    formality_issues: list[dict]
    global_feedback: str
    annotated_text: str
